helix-term/src/compositor.rs

```diff
@@ -132,6 +132,14 @@ impl Compositor {
         self.layers.pop()
     }
 
+    pub fn remove(&mut self, id: &'static str) -> Option<Box<dyn Component>> {
+        let idx = self
+            .layers
+            .iter()
+            .position(|layer| layer.id() == Some(id))?;
```

Comment:
let idx = self.find_id(id);

Reply:
`find_id(id)` returns the actual component that has the specified id. A better name for the function would have been `find_by_id()`.